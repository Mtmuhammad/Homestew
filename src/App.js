import "./Styles/base/_base.scss";
import { Routes, Route } from "react-router-dom";
import Navbar from "./Layout/NavBar/Navbar";
import Footer from "./Layout/Footer/Footer";
import Home from "./Pages/home/Home";
import About from "./Pages/about/About";


function App() {
  return (
    <>
      <Navbar />
      <Routes>
        <Route path="/" element={<App />} />
        <Route index element={<Home />} />
        <Route path="about" element={<About />} />
      </Routes>
      <Footer/>
    </>
  );
}

export default App;
