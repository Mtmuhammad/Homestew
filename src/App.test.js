import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render App without crashing", async () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
  });

  test("should match snapshot", async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(await asFragment()).toMatchSnapshot();
  });
});

// test("should render home component on initial render", async () => {
//   render(
//     <BrowserRouter>
//       <App />
//     </BrowserRouter>
//   );
//   const homeComponent = screen.getByTestId("home");
//   expect(homeComponent).toBeInTheDocument();
// });
