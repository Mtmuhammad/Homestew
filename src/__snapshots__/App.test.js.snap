// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="topbar"
    >
      <div
        class="container d-flex justify-content-between"
      >
        <div
          class="socials"
          style="margin-left: 15px;"
        >
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-brands fa-twitter"
            />
          </a>
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-solid fa-rss"
            />
          </a>
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-brands fa-facebook"
            />
          </a>
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-brands fa-pinterest"
            />
          </a>
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-brands fa-instagram"
            />
          </a>
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-brands fa-youtube"
            />
          </a>
        </div>
        <div
          class="socials"
        >
          <a
            data-testid="user-icon"
            href="/search"
          >
            <i
              class="fa-solid fa-magnifying-glass"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <div
        class="mt-3 mb-1"
      >
        <a
          href="/"
        >
          <img
            alt="Homestew Logo"
            class="d-none d-lg-block logo"
            src="./homestewLogo.svg"
          />
        </a>
      </div>
    </div>
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="Homestew Alt logo"
            class="d-lg-none logo-alt"
            src="/Homestew-alt-logo.svg"
          />
        </a>
        <button
          aria-controls="navbarCollapse"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarCollapse"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="justify-content-center collapse navbar-collapse"
          id="navbarCollapse"
        >
          <ul
            class="navbar-nav mb-2 mb-lg-0"
          >
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="active nav-link"
                href="/"
              >
                HOME
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/about"
              >
                ABOUT
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/search"
              >
                RECIPES
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/diets"
              >
                DIETS
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/blog"
              >
                BLOG
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/contact"
              >
                CONTACT
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <hr />
  </nav>
  <div
    data-testid="home"
  >
    <header
      class="container mt-5"
      style="max-width: 1100px;"
    >
      <div
        class="container pt-3 pb-4"
      >
        <h1
          class="mb-4"
        >
          Featured Recipes
          <i
            class="fa-solid fa-utensils"
          />
        </h1>
        <div
          class="row"
        >
          <div
            class="latest-post col-6 col-md-4"
          >
            <img
              alt="Braised Steak"
              class="img-fluid"
              src="braisedSteak.jpg"
            />
            <div
              class="caption"
            >
              <a
                href="/recipes/635874"
              >
                <p
                  class="caption-text"
                >
                  Braised Steak
                </p>
              </a>
            </div>
          </div>
          <div
            class="latest-post col-6 col-md-4"
          >
            <img
              alt="Simple Caesar Salad"
              class="img-fluid"
              src="caesarSalad.jpg"
            />
            <div
              class="caption"
            >
              <a
                href="/recipes/660085"
              >
                <p
                  class="caption-text"
                >
                  Simple Caesar Salad
                </p>
              </a>
            </div>
          </div>
          <div
            class="latest-post col-md-4 d-none d-md-block"
          >
            <img
              alt="Blueberry Muffins"
              class="img-fluid"
              src="blueberryMuffins.jpg"
            />
            <div
              class="caption"
            >
              <a
                href="/recipes/648647"
              >
                <p
                  class="caption-text"
                >
                  Blueberry Muffins
                </p>
              </a>
            </div>
          </div>
        </div>
      </div>
      <hr
        class="mt-4"
      />
    </header>
    <div
      class="latest-recipes container mt-5"
      style="max-width: 1100px;"
    >
      <div
        class="container"
      >
        <h2
          class="secondary mb-4"
        >
          Latest recipes 
          <i
            class="fa-solid fa-utensils"
          />
        </h2>
        <div
          class="row"
        >
          <div
            class="col-12 col-sm-6 col-lg-4 mb-5"
          >
            <article
              class="text-center post"
            >
              <div
                class="post-img"
              >
                <img
                  alt="Tomato Soup"
                  class="img-fluid"
                  src="tomatoSoup.jpg"
                />
                <div
                  class="category"
                >
                  <span>
                    Gluten Free
                  </span>
                </div>
              </div>
              <div
                class="post-content"
              >
                <h2>
                  Easy Tomato Soup
                </h2>
                <p
                  class="px-5 px-md-3"
                >
                  Easy Tomato Soup is a 
                  <b>
                    gluten free, lacto ovo vegetarian, and primal
                  </b>
                   recipe with 2 servings. One serving contains 
                  <b>
                    299 calories
                  </b>
                  , 
                  <b>
                    4g of protein
                  </b>
                  , and 
                  <b>
                    23g of fat
                  </b>
                  . For 
                  <b>
                    $2.09 per serving
                  </b>
                  , this recipe 
                  <b>
                    covers 14%
                  </b>
                   of your daily requirements of vitamins and minerals...
                </p>
              </div>
              <div
                class="more text-center mt-5"
              >
                <a
                  href="/recipes/1674265"
                >
                  Continue Reading â‡¢
                </a>
              </div>
            </article>
          </div>
          <div
            class="col-12 col-sm-6 col-lg-4 mb-5"
          >
            <article
              class="text-center post"
            >
              <div
                class="post-img"
              >
                <img
                  alt="Crispy Sweet Sticky Spicy Wings"
                  class="img-fluid"
                  src="chickenWings.jpg"
                />
                <div
                  class="category"
                >
                  <span>
                    Appetizer
                  </span>
                </div>
              </div>
              <div
                class="post-content"
              >
                <h2>
                  Sweet Sticky Spicy Wings
                </h2>
                <p
                  class="px-5 px-md-3"
                >
                  The recipe Crispy Sweet Sticky Spicy Wings can be made 
                  <b>
                    in around about 45 minutes
                  </b>
                  . One serving contains 
                  <b>
                    600 calories
                  </b>
                  , 
                  <b>
                    30g of protein
                  </b>
                  , and 
                  <b>
                    24g of fat
                  </b>
                  . This recipe serves 6. For 
                  <b>
                    $1.85 per serving
                  </b>
                  , this recipe 
                  <b>
                    covers 20%
                  </b>
                   of your daily requirements of...
                </p>
              </div>
              <div
                class="more text-center mt-5"
              >
                <a
                  href="/recipes/640841"
                >
                  Continue Reading â‡¢
                </a>
              </div>
            </article>
          </div>
          <div
            class="col-12 col-sm-6 col-lg-4 mb-5"
          >
            <article
              class="text-center post"
            >
              <div
                class="post-img"
              >
                <img
                  alt="black bean and veggie burgers with corn salsa"
                  class="img-fluid"
                  src="veggieBurger.jpg"
                />
                <div
                  class="category"
                >
                  <span>
                    Vegetarian
                  </span>
                </div>
              </div>
              <div
                class="post-content"
              >
                <h2>
                  Black Bean Veggie Burgers
                </h2>
                <p
                  class="px-5 px-md-3"
                >
                  Forget going out to eat or ordering takeout every time you crave American food. Try making Just Julieâ€™s BBQ Spicy Black Bean Burgers at home. Watching your figure? This gluten free and dairy free recipe has 
                  <b>
                    194 calories
                  </b>
                  , 
                  <b>
                    7g of protein
                  </b>
                  , and 
                  <b>
                    8g of fat
                  </b>
                   per serving. This recipe serves...
                </p>
              </div>
              <div
                class="more text-center mt-5"
              >
                <a
                  href="/recipes/648663"
                >
                  Continue Reading â‡¢
                </a>
              </div>
            </article>
          </div>
          <div
            class="col-12 col-sm-6 col-lg-4 mb-5"
          >
            <article
              class="text-center post"
            >
              <div
                class="post-img"
              >
                <img
                  alt="Chicken and Miso Ramen noodle soup"
                  class="img-fluid"
                  src="misoRamen.jpg"
                />
                <div
                  class="category"
                >
                  <span>
                    Chicken
                  </span>
                </div>
              </div>
              <div
                class="post-content"
              >
                <h2>
                  Chicken and Miso Ramen
                </h2>
                <p
                  class="px-5 px-md-3"
                >
                  The recipe Chicken and Miso Ramen Noodle Soup is ready 
                  <b>
                    in approximately 30 minutes
                  </b>
                   and is definitely an awesome 
                  <b>
                    dairy free
                  </b>
                   option for lovers of Japanese food. One serving contains 
                  <b>
                    616 calories
                  </b>
                  , 
                  <b>
                    43g of protein
                  </b>
                  , and 
                  <b>
                    14g of fat
                  </b>
                  . For 
                  <b>
                    $3.46 per serving
                  </b>
                  , this recipe 
                  <b>
                    covers 34%
                  </b>
                   of...
                </p>
              </div>
              <div
                class="more text-center mt-5"
              >
                <a
                  href="/recipes/637908"
                >
                  Continue Reading â‡¢
                </a>
              </div>
            </article>
          </div>
          <div
            class="col-12 col-sm-6 col-lg-4 mb-5"
          >
            <article
              class="text-center post"
            >
              <div
                class="post-img"
              >
                <img
                  alt="Chocolate Chip Cookies"
                  class="img-fluid"
                  src="chocolateChip.jpg"
                />
                <div
                  class="category"
                >
                  <span>
                    Dessert
                  </span>
                </div>
              </div>
              <div
                class="post-content"
              >
                <h2>
                  Chocolate Chip Cookies
                </h2>
                <p
                  class="px-5 px-md-3"
                >
                  Chocolate Chip Cookies might be just the dessert you are searching for. For 
                  <b>
                    $1.52 per serving
                  </b>
                  , this recipe 
                  <b>
                    covers 15%
                  </b>
                   of your daily requirements of vitamins and minerals. This recipe makes 6 servings with 
                  <b>
                    932 calories
                  </b>
                  , 
                  <b>
                    9g of protein
                  </b>
                  , and 
                  <b>
                    49g of fat
                  </b>
                   each. 892 people were impressed...
                </p>
              </div>
              <div
                class="more text-center mt-5"
              >
                <a
                  href="/recipes/637822"
                >
                  Continue Reading â‡¢
                </a>
              </div>
            </article>
          </div>
          <div
            class="col-12 col-sm-6 col-lg-4 mb-5"
          >
            <article
              class="text-center post"
            >
              <div
                class="post-img"
              >
                <img
                  alt="Shrimp Fried Rice"
                  class="img-fluid"
                  src="friedRice.jpg"
                />
                <div
                  class="category"
                >
                  <span>
                    Pescatarian
                  </span>
                </div>
              </div>
              <div
                class="post-content"
              >
                <h2>
                  Shrimp Fried Rice
                </h2>
                <p
                  class="px-5 px-md-3"
                >
                  The recipe Shrimp Fried Rice is ready 
                  <b>
                    in roughly 45 minutes
                  </b>
                   and is definitely a super 
                  <b>
                    gluten free and pescatarian
                  </b>
                   option for lovers of Chinese food. For 
                  <b>
                    $1.36 per serving
                  </b>
                  , you get a main course that serves 4. One serving contains 
                  <b>
                    400 calories
                  </b>
                  , 
                  <b>
                    16g of protein
                  </b>
                  , and 
                  <b>
                    8g of fat
                  </b>
                  . Not a lot...
                </p>
              </div>
              <div
                class="more text-center mt-5"
              >
                <a
                  href="/recipes/682619"
                >
                  Continue Reading â‡¢
                </a>
              </div>
            </article>
          </div>
        </div>
      </div>
    </div>
    <div
      class="icon-bar"
    >
      <div
        class="container"
        style="max-width: 1100px;"
      >
        <div
          class="row"
        >
          <h3
            class="tertiary text-center"
          >
            Explore Recipes:
          </h3>
          <div
            class="col-6 col-lg-3"
          >
            <div
              class="icon text-center"
              data-testid="icon-container"
            >
              <a
                href="/recipes/type/bread"
              >
                <i
                  class="fa-solid fa-bread-slice"
                  data-testid="icon"
                />
                <p>
                  Bread Recipes
                </p>
              </a>
            </div>
          </div>
          <div
            class="col-6 col-lg-3"
          >
            <div
              class="icon text-center"
              data-testid="icon-container"
            >
              <a
                href="/recipes/type/main course"
              >
                <i
                  class="fa-solid fa-drumstick-bite"
                  data-testid="icon"
                />
                <p>
                  Main Course
                </p>
              </a>
            </div>
          </div>
          <div
            class="col-6 col-lg-3"
          >
            <div
              class="icon text-center"
              data-testid="icon-container"
            >
              <a
                href="/recipes/type/side dish"
              >
                <i
                  class="fa-solid fa-bowl-food"
                  data-testid="icon"
                />
                <p>
                  Side Dishes
                </p>
              </a>
            </div>
          </div>
          <div
            class="col-6 col-lg-3"
          >
            <div
              class="icon text-center"
              data-testid="icon-container"
            >
              <a
                href="/recipes/type/dessert"
              >
                <i
                  class="fa-solid fa-cookie"
                  data-testid="icon"
                />
                <p>
                  Desserts
                </p>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="cuisines container"
      style="max-width: 1100px; margin-top: 10rem; margin-bottom: 10rem;"
    >
      <div
        class="container"
      >
        <h3
          class="tertiary"
          style="margin-bottom: 2rem;"
        >
          Spanish recipes 
          <i
            class="fa-solid fa-utensils"
            style="color: rgb(135, 91, 140); font-size: 3rem;"
          />
        </h3>
        <div
          class="row"
        />
        <div
          class="more text-center mt-5"
        >
          <a
            href="/recipes/category/Spanish"
          >
            More Spanish recipes
          </a>
        </div>
      </div>
    </div>
    <div
      class="cuisines container"
      style="max-width: 1100px; margin-top: 10rem; margin-bottom: 10rem;"
    >
      <div
        class="container"
      >
        <h3
          class="tertiary"
          style="margin-bottom: 2rem;"
        >
          Caribbean recipes 
          <i
            class="fa-solid fa-utensils"
            style="color: rgb(0, 128, 128); font-size: 3rem;"
          />
        </h3>
        <div
          class="row"
        />
        <div
          class="more text-center mt-5"
        >
          <a
            href="/recipes/category/Caribbean"
          >
            More Caribbean recipes
          </a>
        </div>
      </div>
    </div>
    <div
      class="cuisines container"
      style="max-width: 1100px; margin-top: 10rem; margin-bottom: 10rem;"
    >
      <div
        class="container"
      >
        <h3
          class="tertiary"
          style="margin-bottom: 2rem;"
        >
          Italian recipes 
          <i
            class="fa-solid fa-utensils"
            style="color: rgb(168, 186, 154); font-size: 3rem;"
          />
        </h3>
        <div
          class="row"
        />
        <div
          class="more text-center mt-5"
        >
          <a
            href="/recipes/category/Italian"
          >
            More Italian recipes
          </a>
        </div>
      </div>
    </div>
    <div
      class="cuisines container"
      style="max-width: 1100px; margin-top: 10rem; margin-bottom: 10rem;"
    >
      <div
        class="container"
      >
        <h3
          class="tertiary"
          style="margin-bottom: 2rem;"
        >
          Indian recipes 
          <i
            class="fa-solid fa-utensils"
            style="color: rgb(135, 91, 140); font-size: 3rem;"
          />
        </h3>
        <div
          class="row"
        />
        <div
          class="more text-center mt-5"
        >
          <a
            href="/recipes/category/Indian"
          >
            More Indian recipes
          </a>
        </div>
      </div>
    </div>
    <div
      class="cuisines container"
      style="max-width: 1100px; margin-top: 10rem; margin-bottom: 10rem;"
    >
      <div
        class="container"
      >
        <h3
          class="tertiary"
          style="margin-bottom: 2rem;"
        >
          Chinese recipes 
          <i
            class="fa-solid fa-utensils"
            style="color: rgb(0, 128, 128); font-size: 3rem;"
          />
        </h3>
        <div
          class="row"
        />
        <div
          class="more text-center mt-5"
        >
          <a
            href="/recipes/category/Chinese"
          >
            More Chinese recipes
          </a>
        </div>
      </div>
    </div>
    <div
      class="cuisines container"
      style="max-width: 1100px; margin-top: 10rem; margin-bottom: 10rem;"
    >
      <div
        class="container"
      >
        <h3
          class="tertiary"
          style="margin-bottom: 2rem;"
        >
          American recipes 
          <i
            class="fa-solid fa-utensils"
            style="color: rgb(168, 186, 154); font-size: 3rem;"
          />
        </h3>
        <div
          class="row"
        />
        <div
          class="more text-center mt-5"
        >
          <a
            href="/recipes/category/American"
          >
            More American recipes
          </a>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="footer"
  >
    <div
      class="container-fluid"
    >
      <div
        class="footer-instagram"
      >
        <div
          class="footer-instagram-avatar"
        >
          <div
            class="footer-instagram-avatar-pic"
          >
            <img
              alt="Instagram Avatar"
              src="avatar.jpg"
            />
          </div>
          <div
            class="footer-instagram-avatar-text"
          >
            <h5>
              @homestew
            </h5>
            <span>
              22,5k followers
            </span>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram1.jpg"
            />
          </div>
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram2.jpg"
            />
          </div>
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram3.jpg"
            />
          </div>
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram4.jpg"
            />
          </div>
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram5.jpg"
            />
          </div>
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram6.jpg"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12"
        >
          <div
            class="footer-text"
          >
            <div
              class="footer-logo"
            >
              <a
                href="/"
              >
                <img
                  alt="Footer alt logo"
                  class="img-fluid mb-4"
                  src="./Homestew-alt-logo.svg"
                />
              </a>
            </div>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Quis ipsum suspendisse ultrices gravida. Risus commodo viverra maecenas accumsan lacus vel facilisis.
            </p>
            <div
              class="footer-social mb-5"
            >
              <a
                href="#0"
              >
                <i
                  class="fa fa-facebook"
                  data-testid="footer-icon"
                />
              </a>
              <a
                href="#0"
              >
                <i
                  class="fa fa-twitter"
                  data-testid="footer-icon"
                />
              </a>
              <a
                href="#0"
              >
                <i
                  class="fa fa-instagram"
                  data-testid="footer-icon"
                />
              </a>
              <a
                href="#0"
              >
                <i
                  class="fa fa-youtube-play"
                  data-testid="footer-icon"
                />
              </a>
              <a
                href="#0"
              >
                <i
                  class="fa fa-envelope-o"
                  data-testid="footer-icon"
                />
              </a>
            </div>
            <div
              class="footer-copyright"
            >
              <p
                class="mb-0"
              >
                Copyright Â© 2022 All rights reserved
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
